crowbar_framework/lib/crowbar/backup/validate.rb

```diff
@@ -0,0 +1,103 @@
+#
+# Copyright 2015, SUSE LINUX Products GmbH
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+module Crowbar
+  class Backup
+    class Validate
+      attr_accessor :path
+
+      def initialize(path)
+        @path = path
+      end
+
+      def validate
+        [:hostname, :version, :file_extension].each do |test|
+          ret = send(test)
+          unless ret[:status] == :ok
+            return ret
+          end
+        end
+
+        { status: :ok, msg: ""}
+      end
+
+      protected
+
+      def hostname
+        hostname_file = File.join(path, "crowbar", "etc", "HOSTNAME")
+        #unless File.exist?(hostname_file)
```

Comment:
Style/CommentIndentation: Incorrect indentation detected (column 8 instead of 10).<br>Style/LeadingCommentSpace: Missing space after #. (https://github.com/bbatsov/ruby-style-guide#hash-space)